[REF] Convert fiscal position rule  partner_fiscal_type_id in many2many

mileo opened the pull request:
WIP

![captura de tela de 2016-04-01 00 13 16](https://cloud.githubusercontent.com/assets/1975978/14196837/919b6b12-f79e-11e5-8f1e-741e751a3cc5.png)

mileo commented:
@renatonlima @danimaribeiro Que que vcs acham?

danimaribeiro commented:
Em relação a funcionalidade :+1:

renatonlima commented:
@mileo Em relação a funcionalidade é bom :+1: , só vou analisar a implementação. você pode dar uma olhada no travis tem um erro de PEP8

mileo commented:
@renatonlima  nem revise que tem coisa errada. 

Foi só um brandstorm, assim que finalizar solicito o label de review.

mileo committed:
[REF] Convert fiscal position rule  partner_fiscal_type_id in many2many

INSERT INTO public.afp_rule_l10n_br_account_partner_fiscal_type(
            afp_id, partner_fiscal_type_id)
     SELECT
        id, partner_fiscal_type_id
     FROM account_fiscal_position_rule
     WHERE partner_fiscal_type_id is not null;

Conflicts:
	l10n_br_account/__openerp__.py
Conflicts:
	l10n_br_account/__openerp__.py

mileo commented:
Needs Review:  @sadamo @leorochael @danimaribeiro @renatonlima

danimaribeiro commented:
Please fix the build before we test, there are two unit tests failling.

mileo committed:
[NEW] Partner Special Type
Conflicts:
	l10n_br_account/models/l10n_br_account.py

renatonlima commented:
@mileo 

Qual é a função do objeto l10n_br_account.partner.special.fiscal.type?

mileo commented:
Criar tipos especiais para beneficos: Isento de icms e ipi por exemplo. (Ele não deixa de ser um contribuinte)

mileo commented:
@renatonlima O que precisa de review?

mbcosta commented on l10n_br_account/models/account_fiscal_position_rule.py:
```diff
@@ -75,12 +101,26 @@ def _map_domain(self, partner, addrs, company, **kwargs):
             ('fiscal_type', '=', company.fiscal_type),
             ('fiscal_category_id', '=', kwargs.get('fiscal_category_id')),
-            '|', ('partner_fiscal_type_id', '=', partner_fiscal_type_id),
+            '|', ('partner_fiscal_type_id', 'in', partner_fiscal_type_id),
```
Acredito que é melhor manter o operador '=' ao invés de troca-lo para 'in' já que o campo recebe um id ( ex.: 1 ) e não uma lista

mbcosta replied:
Acredito que é melhor manter o operador '=' ao invés de troca-lo pelo 'in' já que o campo é um id ( ex.: 1 ) e não uma lista

renatonlima commented:
@mileo da uma olhada na comentário do @mbcosta

mileo committed:
[FIX] The domain term '('partner_fiscal_type_id', 'in', False)' should use the '=' or '!=' operator.

mileo commented:
@renatonlima @mbcosta FIXED
 The domain term '('partner_fiscal_type_id', 'in', False)' should use the '=' or '!=' operator.
